gp-includes/warnings.php

```diff
@@ -815,4 +815,56 @@ private function links_without_url_and_placeholders_are_equal( string $original_
 
 		return $error;
 	}
+
+	/**
+	 * Adds a warning for added or removed spaces at the beginning or at the end of the translation.
+	 *
+	 * @since 4.0.0
+	 * @access public
+	 *
+	 * @param string $original    The original string.
+	 * @param string $translation The translated string.
+	 *
+	 * @return string|true True if check is OK, otherwise warning message.
+	 */
+	public function warning_unexpected_start_end_space( string $original, string $translation ) {
+		$original_start_spaces    = strlen( $original ) - strlen( ltrim( $original, ' ' ) );
+		$original_end_spaces      = strlen( $original ) - strlen( rtrim( $original, ' ' ) );
+		$translation_start_spaces = strlen( $translation ) - strlen( ltrim( $translation, ' ' ) );
+		$translation_end_spaces   = strlen( $translation ) - strlen( rtrim( $translation, ' ' ) );
+
+		$warnings = array();
+		if ( ( $original_start_spaces ) && ( ! $translation_start_spaces ) ) {
```

Comment:
```suggestion
		if ( $original_start_spaces && ! $translation_start_spaces ) {
```

There are few parenthesis below too which can be removed.